### PR TITLE
feat: add suggestions to `use-isnan` in `indexOf` & `lastIndexOf` calls

### LINKED ISSUE #17978: Rule Change: Add suggestions to use-isnan
### What rule do you want to change?

use-isnan

### What change do you want to make?

Implement suggestions

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
value == NaN
```

### What does the rule currently do for this code?

Report an error - and nothing else.

### What will the rule do after it's changed?

Also provide a suggestion to switch to `isNaN(value)`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Either `Number.isNaN` or `isNaN` could be suggested. Maybe the docs should state which is preferred and why?

### PR DESCRIPTION
**Pull request by StyleShit**
#### Prerequisites checklist

- [ X] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[X] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

This PR adds suggestions to `use-isnan` when using `indexOf` and `lastIndexOf`.
The current fix changes them to `findIndex` & `findLastIndex` respectively.
I wanted to suggest also `includes` and `some` as second and third suggestions, but I got into a rabbit hole of checking whether the parent node is a `BinaryExpression` and whether the developer used `indexOf() !== -1`, `indexOf() > 0`, etc.
We can probably do that, just need to find all the cases. Do you think it's worth the effort?

(Probably?) Closes #17978

#### NOTE

Let's wait until https://github.com/eslint/eslint/pull/18059 is merged before merging this one, because it'll require me to extend the new tests that were added there.

### CODE REVIEW

File: lib/rules/use-isnan.js [hunk 1]
```diff
@@ -173,7 +174,34 @@ module.exports = {
                     node.arguments.length === 1 &&
                     isNaNIdentifier(node.arguments[0])
                 ) {
-                    context.report({ node, messageId: "indexOfNaN", data: { methodName } });
+
+                    /*
+                     * We can't fix things like `arr.indexOf((doStuff(), NaN))` because we'll need to somehow call the expressions
+                     * before `NaN` in order to preserve side effects, which is not possible with fixes like `arr.findIndex(Number.isNaN)`.
```
Comment (amareshsm) [inline lib/rules/use-isnan.js:180]:
```suggestion
                     * To retain side effects, it's essential to address NaN beforehand, which is not possible with fixes like `arr.findIndex(Number.isNaN)`.
```

**Comment by nzakas** [comment]
Marking as a draft so we don't accidentally merge.

**Comment by StyleShit** [comment]
@nzakas Ready for review 😄

File: lib/rules/use-isnan.js [hunk 2]
```diff
@@ -184,7 +185,35 @@ module.exports = {
                     node.arguments.length === 1 &&
                     isNaNIdentifier(node.arguments[0])
                 ) {
-                    context.report({ node, messageId: "indexOfNaN", data: { methodName } });
+
+                    /*
+                     * To retain side effects, it's essential to address `NaN` beforehand, which
+                     * is not possible with fixes like `arr.findIndex(Number.isNaN)`.
+                     */
+                    const isFixable = node.arguments[0].type !== "SequenceExpression";
```
Comment (nzakas) [inline lib/rules/use-isnan.js:193]: I think `isFixable` actually means `isSuggestable`? (To disambiguate between fixes and suggestions, it's probably better to keep the naming in line.)

**Review by nzakas — CHANGES_REQUESTED** [review]
Overall LGTM, just a question around naming.

File: lib/rules/use-isnan.js (hunk 2 above)
Comment (StyleShit) [reply lib/rules/use-isnan.js]: ugh... I always mix them up...
fixed (pun intended 😄)

File: lib/rules/use-isnan.js [hunk 3]
```diff
@@ -184,7 +185,35 @@ module.exports = {
...
+
+                    /*
+                     * To retain side effects, it's essential to address `NaN` beforehand, which
+                     * is not possible with fixes like `arr.findIndex(Number.isNaN)`.
+                     */
+                    const isSuggestable = node.arguments[0].type !== "SequenceExpression";
+                    const suggestedFixes = [];
+
+                    if (isSuggestable) {
+                        const shouldWrap = callee.property.type === "Literal";
```
Comment (mdjermanovic) [inline lib/rules/use-isnan.js:197]:
```suggestion
                        const shouldWrap = callee.computed;
```

Property can also be a template literal, for example::

```js
var firstIndex = myArray[`indexOf`](NaN);
```

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks! Leaving open for @nzakas to verify.

File: lib/rules/use-isnan.js (hunk 3 above)
Comment (StyleShit) [reply lib/rules/use-isnan.js]: Nice catch!

**Review by nzakas — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**